### PR TITLE
Metrics Engine v2

### PR DESCRIPTION
**Pull request by pastorsx**
Real time measurements through Meep-Metrics-Engine service

-implementation of v2 endpoints for getting the real time metrics for latency, throughput, packet loss, events
-support for former metrics engine (now called v1)

TESTING
- scenario ran, queried, response tested using curl and browser

### CODE REVIEW

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,790 @@
...
+          \ when retrying the request allows the operator to avoid creating a duplicate\
+          \ subscription."
+      callbackReference:
+        $ref: "#/definitions/Events_callbackReference"
+      eventQueryParams:
+        $ref: "#/definitions/EventQueryParams"
+      recurrence:
+        type: "integer"
+        example: 1
+        description: "Recurrence of recurring-time-based notifications"
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename property: updateInterval

Description: "Notification interval in seconds, disabled if set to 0"

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+      callbackReference:
+        notifyURL: "http://clientApp.example.com/metric_event_notifications/123456"
+      eventQueryParams:
+        tags:
+        - name: "type"
+          value: "MOBILITY"
+        fields:
+        - "event"
+      recurrence: 1
+      subscriptionType: "Recurring-time-based"
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Adjust with new names

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+      recurrence:
+        type: "integer"
+        example: 1
+        description: "Recurrence of recurring-time-based notifications"
+      subscriptionType:
+        type: "string"
+        example: "Recurring-time-based"
+        description: "Type of subscription triggering notifications"
+        enum:
+        - "Recurring-time-based"
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename property: updateInterval

Description: "Notification interval in seconds, disabled if set to 0"

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 4]
```diff
@@ -0,0 +1,790 @@
...
+        fields:
+        - "event"
+        tags:
+        - name: "type"
+          value: "MOBILITY"
+      callbackReference: "{\"notifyURL\":\"http://clientApp.example.com/metric_event_notifications/123456\"\
+        }"
+      clientCorrelator: "83"
+      subscriptionId: "1.2556487E7"
+  EventSubscriptionResponseList:
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename definition: EventSubscriptionList

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+    properties:
+      eventSubscriptionResponse:
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename property: eventSubscriptions

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 6]
```diff
@@ -0,0 +1,790 @@
...
+      recurrence:
+        type: "integer"
+        example: 1
+        description: "Recurrence of recurring-time-based notifications"
+      subscriptionType:
+        type: "string"
+        example: "Recurring-time-based"
+        description: "Type of subscription triggering notifications"
+        enum:
+        - "Recurring-time-based"
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename property: updateInterval

Description: "Notification interval in seconds, disabled if set to 0"

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 7]
```diff
@@ -0,0 +1,790 @@
...
+          value: "ue1-iperf"
+        fields:
+        - "lat"
+        - "tput"
+        - "loss"
+        scope:
+          limit: 60
+      recurrence: 1
+      subscriptionType: "Recurring-time-based"
+  NetworkSubscriptionResponse:
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename definition: NetworkSubscription

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 8]
```diff
@@ -0,0 +1,790 @@
...
+      recurrence:
+        type: "integer"
+        example: 1
+        description: "Recurrence of recurring-time-based notifications"
+      subscriptionType:
+        type: "string"
+        example: "Recurring-time-based"
+        description: "Type of subscription triggering notifications"
+        enum:
+        - "Recurring-time-based"
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename property: updateInterval

Description: "Notification interval in seconds, disabled if set to 0"

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 9]
```diff
@@ -0,0 +1,790 @@
...
+        tags:
+        - name: "src"
+          value: "ue1-iperf"
+        - name: "dest"
+          value: "zone1-fog1-iperf"
+      callbackReference: "{\"notifyURL\":\"http://clientApp.example.com/metric_network_notifications/123456\"\
+        }"
+      clientCorrelator: "83"
+      subscriptionId: "1.2556487E7"
+  NetworkSubscriptionResponseList:
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename definition: NetworkSubscriptionList

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 10]
```diff
@@ -0,0 +1,790 @@
...
+    properties:
+      networkSubscriptionResponse:
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/swagger-v2.yaml]: Rename property: networkSubscriptions

File: go-apps/meep-metrics-engine/server/api_metrics.go [hunk 11]
```diff
@@ -31,3 +31,11 @@ import (
 func MetricsGet(w http.ResponseWriter, r *http.Request) {
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/api_metrics.go]: move generated code to version-specific folders

File: go-apps/meep-metrics-engine/server/api_subscriptions.go [hunk 12]
```diff
@@ -0,0 +1,61 @@
+/*
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/api_subscriptions.go]: move generated code to version-specific folders

File: go-apps/meep-metrics-engine/server/convert.go [hunk 13]
```diff
@@ -0,0 +1,96 @@
+/*
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/convert.go]: move code to version-specific folders

File: go-apps/meep-metrics-engine/server/metrics-engine.go [hunk 14]
```diff
@@ -19,15 +19,67 @@ package server
 import (
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/metrics-engine.go]: Separate code into version-specific folders.
Remove version references.
Adapt code to object name changes.

File: go-apps/meep-metrics-engine/server/routers.go [hunk 15]
```diff
@@ -58,7 +58,11 @@ func NewRouter() *mux.Router {
 }
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/routers.go]: Route requests to correct version-specific functions.
Update function names after code generation.

File: .meepctl-repocfg.yaml [hunk 16]
```diff
@@ -47,6 +47,7 @@ repo:
       api-bundle:
         - meep-ctrl-engine
         - meep-metrics-engine
+        - meep-metrics-engine-v2-notification
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply .meepctl-repocfg.yaml]: fixed

File: .meepctl-repocfg.yaml [hunk 17]
```diff
@@ -94,7 +95,7 @@ repo:
       codecov: true
       lint: true
       nodeport: 30008
-      api: go-apps/meep-metrics-engine/api/swagger.yaml
+      api: go-apps/meep-metrics-engine/api/swagger-v2.yaml
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply .meepctl-repocfg.yaml]: fixed

File: .meepctl-repocfg.yaml [hunk 18]
```diff
@@ -117,6 +118,8 @@ repo:
       api: go-packages/meep-loc-serv-notification-client/api/swagger.yaml
     meep-mg-app-notification:
       api: go-packages/meep-mg-app-client/api/swagger.yaml
+    meep-metrics-engine-v2-notification:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply .meepctl-repocfg.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 19]
```diff
@@ -0,0 +1,790 @@
...
+          fields:
+          - "event"
+          tags:
+          - name: "type"
+            value: "MOBILITY"
+        callbackReference: "{\"notifyURL\":\"http://clientApp.example.com/metric_event_notifications/123456\"\
+          }"
+        clientCorrelator: "83"
+        subscriptionId: "1.2556487E7"
+  Events_callbackReference:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 20]
```diff
@@ -0,0 +1,790 @@
...
+          tags:
+          - name: "src"
+            value: "ue1-iperf"
+          - name: "dest"
+            value: "zone1-fog1-iperf"
+        callbackReference: "{\"notifyURL\":\"http://clientApp.example.com/metric_network_notifications/123456\"\
+          }"
+        clientCorrelator: "83"
+        subscriptionId: "1.2556487E7"
+  Network_callbackReference:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 21]
```diff
@@ -0,0 +1,790 @@
...
+    x-exportParamName: "StartTime"
+    x-optionalDataType: "String"
+  Query.StopTime:
+    name: "stopTime"
+    in: "query"
+    description: "Ending timestamp of time range query"
+    required: false
+    type: "string"
+    x-exportParamName: "StopTime"
+    x-optionalDataType: "String"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 22]
```diff
@@ -0,0 +1,790 @@
...
+- name: "Metrics"
+- name: "Subscriptions"
+schemes:
+- "http"
+consumes:
+- "application/json"
+produces:
+- "application/json"
+paths:
+  /metrics/network:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 23]
```diff
@@ -0,0 +1,790 @@
+---
+swagger: "2.0"
+info:
+  description: "Metrics Service provides metrics about the active scenario <p>**Micro-service**<br>[meep-metrics-engine](https://github.com/InterDigitalInc/AdvantEDGE/tree/master/go-apps/meep-metrics-engine)\
+    \ <p>**Type & Usage**<br>Platform Service used by control/monitoring software\
+    \ and possibly by edge applications that require metrics <p>**Details**<br>API\
+    \ details available at _your-AdvantEDGE-ip-address:30000/api_ <p>**Default Port**<br>`30005`"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 24]
```diff
@@ -0,0 +1,790 @@
...
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/NetworkQueryResponse"
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+  /subscriptions/network:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 25]
```diff
@@ -0,0 +1,790 @@
...
+      description: "Returns all Network Metric subscriptions"
+      operationId: "getNetworkMetricSubscription"
+      produces:
+      - "application/json"
+      parameters: []
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/NetworkSubscriptionResponseList"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 26]
```diff
@@ -0,0 +1,790 @@
...
+          $ref: "#/definitions/NetworkSubscriptionParams"
+        x-exportParamName: "Params"
+      responses:
+        201:
+          description: "Created"
+          schema:
+            $ref: "#/definitions/NetworkSubscriptionResponse"
+        500:
+          description: "Internal server error"
+  /subscriptions/network/{subscriptionId}:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 27]
```diff
@@ -0,0 +1,790 @@
...
+        type: "string"
+        x-exportParamName: "SubscriptionId"
+      responses:
+        204:
+          description: "deleted"
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+  /metrics/event:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 28]
```diff
@@ -0,0 +1,790 @@
...
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/EventQueryResponse"
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+  /subscriptions/event:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 29]
```diff
@@ -0,0 +1,790 @@
...
+          $ref: "#/definitions/EventSubscriptionParams"
+        x-exportParamName: "Params"
+      responses:
+        201:
+          description: "Created"
+          schema:
+            $ref: "#/definitions/EventSubscriptionResponse"
+        500:
+          description: "Internal server error"
+  /subscriptions/event/{subscriptionId}:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 30]
```diff
@@ -0,0 +1,790 @@
...
+      description: "Returns all Event Metric subscriptions"
+      operationId: "getEventMetricSubscription"
+      produces:
+      - "application/json"
+      parameters: []
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/EventSubscriptionResponseList"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 31]
```diff
@@ -0,0 +1,790 @@
...
+        in: "path"
+        description: "Subscription ID - returned when the subscription was created"
+        required: true
+        type: "string"
+        x-exportParamName: "SubscriptionId"
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/NetworkSubscriptionResponse"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 32]
```diff
@@ -0,0 +1,790 @@
...
+        description: "Network subscription parameters"
+        required: true
+        schema:
+          $ref: "#/definitions/NetworkSubscriptionParams"
+        x-exportParamName: "Params"
+      responses:
+        201:
+          description: "Created"
+          schema:
+            $ref: "#/definitions/NetworkSubscriptionResponse"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 33]
```diff
@@ -0,0 +1,790 @@
...
+- "application/json"
+produces:
+- "application/json"
+paths:
+  /metrics/network:
+    post:
+      tags:
+      - "Metrics"
+      description: "Returns Network metrics according to specificed parameters"
+      operationId: "getNetworkMetrics"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 34]
```diff
@@ -0,0 +1,790 @@
...
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+  /subscriptions/network:
+    get:
+      tags:
+      - "Subscriptions"
+      description: "Returns all Network Metric subscriptions"
+      operationId: "getNetworkMetricSubscription"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 35]
```diff
@@ -0,0 +1,790 @@
...
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+  /metrics/event:
+    post:
+      tags:
+      - "Metrics"
+      description: "Returns Event metrics according to specificed parameters"
+      operationId: "getEventMetrics"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 36]
```diff
@@ -0,0 +1,790 @@
...
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+  /subscriptions/event:
+    get:
+      tags:
+      - "Subscriptions"
+      description: "Returns all Event Metric subscriptions"
+      operationId: "getEventMetricSubscription"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 37]
```diff
@@ -0,0 +1,790 @@
...
+          description: "OK"
+          schema:
+            $ref: "#/definitions/NetworkSubscriptionResponseList"
+        500:
+          description: "Internal server error"
+    post:
+      tags:
+      - "Subscriptions"
+      description: "Create a Network Metric subscription"
+      operationId: "createNetworkMetricsSubscription"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 38]
```diff
@@ -0,0 +1,790 @@
...
+          schema:
+            $ref: "#/definitions/NetworkSubscriptionResponse"
+        500:
+          description: "Internal server error"
+  /subscriptions/network/{subscriptionId}:
+    get:
+      tags:
+      - "Subscriptions"
+      description: "Returns a Network Metric subscription"
+      operationId: "getNetworkMetricSubscriptionById"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 39]
```diff
@@ -0,0 +1,790 @@
...
+            $ref: "#/definitions/NetworkSubscriptionResponse"
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+    delete:
+      tags:
+      - "Subscriptions"
+      description: "Returns a Network Metric subscription"
+      operationId: "deleteNetworkMetricSubscriptionById"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 40]
```diff
@@ -0,0 +1,790 @@
...
+          description: "OK"
+          schema:
+            $ref: "#/definitions/EventSubscriptionResponseList"
+        500:
+          description: "Internal server error"
+    post:
+      tags:
+      - "Subscriptions"
+      description: "Create a Event Metric subscription"
+      operationId: "createEventsMetricsSubscription"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 41]
```diff
@@ -0,0 +1,790 @@
...
+        description: "Event subscription parameters"
+        required: true
+        schema:
+          $ref: "#/definitions/EventSubscriptionParams"
+        x-exportParamName: "Params"
+      responses:
+        201:
+          description: "Created"
+          schema:
+            $ref: "#/definitions/EventSubscriptionResponse"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 42]
```diff
@@ -0,0 +1,790 @@
...
+        500:
+          description: "Internal server error"
+  /subscriptions/event/{subscriptionId}:
+    get:
+      tags:
+      - "Subscriptions"
+      description: "Returns an Event Metric subscription"
+      operationId: "getEventMetricSubscriptionById"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 43]
```diff
@@ -0,0 +1,790 @@
...
+            $ref: "#/definitions/EventSubscriptionResponse"
+        404:
+          description: "Not found"
+        500:
+          description: "Internal server error"
+    delete:
+      tags:
+      - "Subscriptions"
+      description: "Returns an Event Metric subscription"
+      operationId: "deleteEventMetricSubscriptionById"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 44]
```diff
@@ -0,0 +1,790 @@
...
+        in: "path"
+        description: "Subscription ID - returned when the subscription was created"
+        required: true
+        type: "string"
+        x-exportParamName: "SubscriptionId"
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/EventSubscriptionResponse"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 45]
```diff
@@ -0,0 +1,790 @@
...
+    type: "object"
+    properties:
+      limit:
+        type: "integer"
+        example: 60
+        description: "Maximum number of points in the response"
+      duration:
+        type: "string"
+        example: "10s"
+        description: "Lenght of time range (specify s for seconds, m for minutes,\
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 46]
```diff
@@ -0,0 +1,790 @@
...
+        description: "Query parameters"
+        required: true
+        schema:
+          $ref: "#/definitions/NetworkQueryParams"
+        x-exportParamName: "Params"
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/NetworkQueryResponse"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 47]
```diff
@@ -0,0 +1,790 @@
...
+        description: "Query parameters"
+        required: true
+        schema:
+          $ref: "#/definitions/EventQueryParams"
+        x-exportParamName: "Params"
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/EventQueryResponse"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 48]
```diff
@@ -0,0 +1,790 @@
...
+      duration:
+        type: "string"
+        example: "10s"
+        description: "Lenght of time range (specify s for seconds, m for minutes,\
+          \ d for days)"
+    description: "Query scope (time / number of points)"
+    example:
+      duration: "10s"
+      limit: 60
+  NetworkValue:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 49]
```diff
@@ -0,0 +1,790 @@
...
+      fields:
+      - "lat"
+      - "tput"
+      - "loss"
+      tags:
+      - name: "src"
+        value: "ue1-iperf"
+      - name: "dest"
+        value: "zone1-fog1-iperf"
+  NetworkQueryResponse:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 50]
```diff
@@ -0,0 +1,790 @@
...
+        notifyURL: "http://clientApp.example.com/metric_event_notifications/123456"
+      eventQueryParams:
+        tags:
+        - name: "type"
+          value: "MOBILITY"
+        fields:
+        - "event"
+      recurrence: 1
+      subscriptionType: "Recurring-time-based"
+  EventSubscriptionResponse:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 51]
```diff
@@ -0,0 +1,790 @@
...
+        description: "Time of event value"
+      description:
+        type: "string"
+        example: "Mobility event ue1 to poa3"
+        description: "Event description"
+    description: "Value of a single event metric"
+    example:
+      description: "Mobility event ue1 to poa3"
+      time: "2019-11-24T12:45:00-5:00"
+  EventQueryResponse:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 52]
```diff
@@ -0,0 +1,790 @@
...
+    example:
+      scope:
+        duration: "10s"
+        limit: 60
+      fields:
+      - "event"
+      tags:
+      - name: "type"
+        value: "MOBILITY"
+  EventValue:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 53]
```diff
@@ -0,0 +1,790 @@
...
+    type: "object"
+    properties:
+      tags:
+        type: "array"
+        description: "Tag array<br>combination of: target, type"
+        items:
+          $ref: "#/definitions/Tag"
+      fields:
+        type: "array"
+        description: "Field array<br>combination of: lat, tput, loss"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 54]
```diff
@@ -0,0 +1,790 @@
...
+        time: "2019-11-24T12:45:00-5:00"
+        tput: 100000.0
+        lat: 5.123
+      name: "network metrics"
+  EventQueryParams:
+    type: "object"
+    properties:
+      tags:
+        type: "array"
+        description: "Tag array<br>combination of: target, type"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 55]
```diff
@@ -0,0 +1,790 @@
...
+      - name: "type"
+        value: "MOBILITY"
+  EventValue:
+    type: "object"
+    properties:
+      time:
+        type: "string"
+        example: "2019-11-24T12:45:00-5:00"
+        description: "Time of event value"
+      description:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 56]
```diff
@@ -0,0 +1,790 @@
...
+          $ref: "#/definitions/EventValue"
+    description: "Event metrics query response"
+    example:
+      columns:
+      - "event"
+      values:
+      - description: "Mobility event ue1 to poa3"
+        time: "2019-11-24T12:45:00-5:00"
+      - description: "Mobility event ue1 to poa2"
+        time: "2019-11-24T12:44:00-5:00"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 4 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 5 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 3 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 6 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 8 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 10 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 9 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 7 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 1 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/server/routers.go (hunk 15 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/server/routers.go]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml (hunk 2 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: fixed

File: go-apps/meep-metrics-engine/server/api_metrics.go (hunk 11 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/server/api_metrics.go]: fixed

File: go-apps/meep-metrics-engine/server/api_subscriptions.go (hunk 12 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/server/api_subscriptions.go]: fixed

File: go-apps/meep-metrics-engine/server/convert.go (hunk 13 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/server/convert.go]: fixed

File: go-apps/meep-metrics-engine/server/metrics-engine.go (hunk 14 above)
Comment (pastorsx) [reply go-apps/meep-metrics-engine/server/metrics-engine.go]: fixed

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 57]
```diff
@@ -0,0 +1,790 @@
...
+        example: "10s"
+        description: "Lenght of time range (specify s for seconds, m for minutes,\
+          \ d for days)"
+    description: "Query scope (time / number of points)"
+    example:
+      duration: "10s"
+      limit: 60
+  NetworkValue:
+    type: "object"
+    properties:
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: not updated until code is changed to support it

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 58]
```diff
@@ -0,0 +1,790 @@
...
+      tags:
+        type: "array"
+        description: "Tag array<br>combination of: src, dest, srcpl, dstpl"
+        items:
+          $ref: "#/definitions/Tag"
+      fields:
+        type: "array"
+        description: "Field array<br>combination of: lat, tput, loss"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: not updated until code is changed to support it

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 59]
```diff
@@ -0,0 +1,790 @@
...
+      values:
+      - loss: 0.003
+        time: "2019-11-24T12:45:00-5:00"
+        tput: 100000.0
+        lat: 5.123
+      - loss: 0.003
+        time: "2019-11-24T12:45:00-5:00"
+        tput: 100000.0
+        lat: 5.123
+      name: "network metrics"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: not updated until code is changed to support it

File: go-apps/meep-metrics-engine/api/swagger-v2.yaml [hunk 60]
```diff
@@ -0,0 +1,790 @@
...
+      recurrence:
+        type: "integer"
+        example: 1
+        description: "Recurrence of recurring-time-based notifications"
+      subscriptionType:
+        type: "string"
+        example: "Recurring-time-based"
+        description: "Type of subscription triggering notifications"
+        enum:
+        - "Recurring-time-based"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-metrics-engine/api/swagger-v2.yaml]: kept it, changed it to Periodic

**Pull request merged**